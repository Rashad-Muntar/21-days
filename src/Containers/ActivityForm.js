import PropTypes from 'prop-types';
import '../assets/styles/ActivityForm.css';

const ActivityForm = ({ handleChangeTitle, handleSubmit }) => (
  <div className="new-activity">
    <i className="clock outline icon" />
    <p>Create a new Activity</p>
    <small>Start the timer of the activity you are about to take</small>
    <form onSubmit={handleSubmit}>
      <input onChange={handleChangeTitle} type="text" placeholder="Enter your next Activity" />
      <button type="submit">Submit</button>
    </form>
  </div>
);

ActivityForm.propTypes = {
  handleChangeTitle: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default ActivityForm;
