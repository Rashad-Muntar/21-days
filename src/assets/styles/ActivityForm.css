.new-activity{
    background-color: #3a062d;
    height: auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 300px;
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
    width: 50%;
}
.new-activity form{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80%;
  margin: 0 auto;
}

.new-activity p{
    color: wheat;
  }

  .new-activity button{
    color: wheat;
    background-color: wheat;
    color: #3a062d;
    border: none;
    padding: 10px 20px;
    border-radius: 7px;
  }

  .new-activity small{
    color: wheat;
    margin-bottom: 20px;
  }

  .new-activity i.icon.outline{
    font-size: 4em;
    color: wheat;
    margin-bottom: 30px;
}


@media screen and (max-width: 460px) {
    .new-activity{
        width: 100%;
        height: 280px;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
        border: 1px solid gray;
    }
    .new-activity p{
        color: wheat;
      }

      .new-activity i.icon.outline{
        font-size: 4em;
        color: wheat;
        margin-bottom: 20px;
    }
}