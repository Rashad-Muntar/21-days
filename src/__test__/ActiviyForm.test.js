import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ActivityForm from '../Containers/ActivityForm';

describe('Home', () => {
  beforeEach(() => {
    render(<ActivityForm />);
  });

  it('Renders Activity form component', () => {
    const tree = renderer.create(<ActivityForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
