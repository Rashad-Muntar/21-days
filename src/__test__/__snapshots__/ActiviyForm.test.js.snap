// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders Activity form component 1`] = `
<div
  className="new-activity"
>
  <i
    className="clock outline icon"
  />
  <p>
    Create a new Activity
  </p>
  <small>
    Start the timer of the activity you are about to take
  </small>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Enter your next Activity"
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
